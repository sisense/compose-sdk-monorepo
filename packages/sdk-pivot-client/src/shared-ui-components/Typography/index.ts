export { Typography, type TypographyProps } from './Typography';
export { typographyTheme, type TypographyTheme } from './themes';
