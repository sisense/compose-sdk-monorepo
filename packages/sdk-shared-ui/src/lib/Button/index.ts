import Button from './Button';

export default Button;
export * from './Button';
export { buttonsTheme, type ButtonsTheme } from './themes';
