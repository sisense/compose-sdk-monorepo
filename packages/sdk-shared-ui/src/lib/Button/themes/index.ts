export { buttonsTheme, type ButtonsTheme } from './buttonsTheme';
