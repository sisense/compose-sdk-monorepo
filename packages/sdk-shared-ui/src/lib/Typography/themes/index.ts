export { typographyTheme, type TypographyTheme } from './typographyTheme';
