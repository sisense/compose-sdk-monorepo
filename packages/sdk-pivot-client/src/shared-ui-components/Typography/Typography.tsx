import React, { ForwardedRef } from 'react';
import {
  Typography as MuiTypography,
  type TypographyProps as MuiTypographyProps,
} from '@mui/material';
import { ThemeProvider } from '@mui/material/styles';

import { typographyTheme, type TypographyTheme } from './themes';

export type TypographyProps = MuiTypographyProps & {
  theme?: TypographyTheme;
  dataTestId?: string;
} & React.HTMLAttributes<HTMLSpanElement>;

export const Typography = React.forwardRef(
  (
    { children, theme = typographyTheme, dataTestId, ...rest }: TypographyProps,
    ref: ForwardedRef<HTMLSpanElement>,
  ) => (
    <ThemeProvider theme={theme}>
      <MuiTypography ref={ref} {...rest} data-testid={dataTestId}>
        {children}
      </MuiTypography>
    </ThemeProvider>
  ),
);
