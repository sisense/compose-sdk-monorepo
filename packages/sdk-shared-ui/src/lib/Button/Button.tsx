import React, { type ForwardedRef, forwardRef } from 'react';
import { Button as MuiButton, ButtonProps as MuiButtonProps } from '@mui/material';
import { ThemeProvider } from '@mui/material/styles';
import { deepmerge } from '@mui/utils';

import { type ButtonsTheme, buttonsTheme } from './themes';

export type ButtonProps = {
  theme?: ButtonsTheme;
} & MuiButtonProps &
  React.ButtonHTMLAttributes<HTMLButtonElement>;

const Button = forwardRef((props: ButtonProps, ref: ForwardedRef<HTMLButtonElement>) => {
  const { children, theme, ...rest } = props;

  const appliedButtonTheme = theme
    ? deepmerge(buttonsTheme, {
        components: {
          MuiButton: theme,
        },
      })
    : buttonsTheme;

  return (
    <ThemeProvider theme={appliedButtonTheme}>
      <MuiButton {...rest} ref={ref}>
        {children}
      </MuiButton>
    </ThemeProvider>
  );
});

export default Button;
export { Button };
