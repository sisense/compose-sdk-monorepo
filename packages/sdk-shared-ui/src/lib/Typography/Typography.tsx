import React, { ForwardedRef } from 'react';
import MuiTypography, {
  type TypographyProps as MuiTypographyProps,
} from '@mui/material/Typography';
import ThemeProvider from '@mui/material/styles/ThemeProvider';

import { typographyTheme, type TypographyTheme } from './themes';

export type TypographyProps = MuiTypographyProps & {
  theme?: TypographyTheme;
  dataTestId?: string;
} & React.HTMLAttributes<HTMLSpanElement>;

const Typography = React.forwardRef(
  (
    { children, theme = typographyTheme, dataTestId, ...rest }: TypographyProps,
    ref: ForwardedRef<HTMLSpanElement>,
  ) => (
    <ThemeProvider theme={theme}>
      <MuiTypography ref={ref} {...rest} data-testid={dataTestId}>
        {children}
      </MuiTypography>
    </ThemeProvider>
  ),
);

export default Typography;
export { Typography };
