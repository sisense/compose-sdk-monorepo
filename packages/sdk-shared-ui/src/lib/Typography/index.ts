import Typography from './Typography';

export default Typography;
export * from './Typography';
export { typographyTheme, type TypographyTheme } from './themes';
