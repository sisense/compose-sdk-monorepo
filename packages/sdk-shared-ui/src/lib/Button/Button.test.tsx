import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { describe, it, expect, vi } from 'vitest';

import { Typography } from '../Typography';
import { Button } from './Button';

import '@testing-library/jest-dom';

describe('Button Component', () => {
  it('should render the Button with default props', () => {
    render(<Button variant="text">Click Me</Button>);

    const button = screen.getByRole('button', { name: 'Click Me' });

    expect(button).toBeInTheDocument();
  });

  it('should match the snapshot when variant gray', () => {
    render(<Button variant="gray">Variant Gray</Button>);

    const button = screen.getByRole('button', { name: 'Variant Gray' });

    expect(button).toMatchSnapshot();
  });

  it('should match the snapshot when variant primary', () => {
    render(<Button variant="primary">Variant Primary</Button>);

    const button = screen.getByRole('button', { name: 'Variant Primary' });

    expect(button).toMatchSnapshot();
  });

  it('should match the snapshot when variant secondary', () => {
    render(<Button variant="secondary">Variant Secondary</Button>);

    const button = screen.getByRole('button', { name: 'Variant Secondary' });

    expect(button).toMatchSnapshot();
  });

  it('should match the snapshot when variant text', () => {
    render(<Button variant="text">Variant Text</Button>);

    const button = screen.getByRole('button', { name: 'Variant Text' });

    expect(button).toMatchSnapshot();
  });

  it('should apply custom themes correctly', () => {
    const customTheme = {
      styleOverrides: {
        root: {
          backgroundColor: 'rgb(0, 0, 255)',
        },
      },
    };

    render(
      <Button theme={customTheme} variant="text">
        Custom Theme
      </Button>,
    );
    const button = screen.getByRole('button', { name: 'Custom Theme' });

    expect(button).toHaveStyle('background-color: rgb(0, 0, 255)');
  });

  it('should use the default buttons theme when no theme is provided', () => {
    render(<Button>Default Theme</Button>);
    const button = screen.getByRole('button', { name: 'Default Theme' });

    expect(button).toHaveStyle('background-color: ButtonFace');
  });

  it('should pass additional props to the MUI Button', () => {
    render(
      <Button type="submit" variant="text">
        Submit
      </Button>,
    );

    const button = screen.getByRole('button', { name: 'Submit' });

    expect(button).toHaveAttribute('type', 'submit');
  });

  it('should handle click events', () => {
    const onClick = vi.fn();
    render(
      <Button onClick={onClick} variant="text">
        Click Event Handler
      </Button>,
    );
    const button = screen.getByRole('button', { name: 'Click Event Handler' });

    fireEvent.click(button);

    expect(onClick).toHaveBeenCalledTimes(1);
  });

  it('should forward refs correctly', () => {
    const ref = React.createRef<HTMLButtonElement>();
    render(
      <Button ref={ref} variant="text">
        With Ref
      </Button>,
    );

    expect(ref.current).not.toBeNull();
    expect(ref.current?.tagName).toBe('BUTTON');
  });

  it('should render with a custom className', () => {
    render(
      <Button className="custom-class" variant="text">
        Styled Button
      </Button>,
    );

    const button = screen.getByRole('button', { name: 'Styled Button' });

    expect(button).toHaveClass('custom-class');
  });

  it('should render children correctly', async () => {
    render(
      <Button className="custom-class" variant="text">
        <Typography variant={'bodyUI'}>Children</Typography>
      </Button>,
    );

    expect(await screen.findByText('Children')).toBeInTheDocument();
  });
});
