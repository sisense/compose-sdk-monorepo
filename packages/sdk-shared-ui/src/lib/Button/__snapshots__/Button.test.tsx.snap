// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button Component > should match the snapshot when variant gray 1`] = `
<button
  class="MuiButtonBase-root MuiButton-root MuiButton-gray MuiButton-grayPrimary MuiButton-sizeMedium MuiButton-graySizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-gray MuiButton-grayPrimary MuiButton-sizeMedium MuiButton-graySizeMedium MuiButton-colorPrimary css-nlk8yv-MuiButtonBase-root-MuiButton-root"
  tabindex="0"
  type="button"
>
  Variant Gray
</button>
`;

exports[`Button Component > should match the snapshot when variant primary 1`] = `
<button
  class="MuiButtonBase-root MuiButton-root MuiButton-primary MuiButton-primaryPrimary MuiButton-sizeMedium MuiButton-primarySizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-primary MuiButton-primaryPrimary MuiButton-sizeMedium MuiButton-primarySizeMedium MuiButton-colorPrimary css-1fw694k-MuiButtonBase-root-MuiButton-root"
  tabindex="0"
  type="button"
>
  Variant Primary
</button>
`;

exports[`Button Component > should match the snapshot when variant secondary 1`] = `
<button
  class="MuiButtonBase-root MuiButton-root MuiButton-secondary MuiButton-secondaryPrimary MuiButton-sizeMedium MuiButton-secondarySizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-secondary MuiButton-secondaryPrimary MuiButton-sizeMedium MuiButton-secondarySizeMedium MuiButton-colorPrimary css-rhvemw-MuiButtonBase-root-MuiButton-root"
  tabindex="0"
  type="button"
>
  Variant Secondary
</button>
`;

exports[`Button Component > should match the snapshot when variant text 1`] = `
<button
  class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1rk2iuf-MuiButtonBase-root-MuiButton-root"
  tabindex="0"
  type="button"
>
  Variant Text
</button>
`;
